formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/FormstackSarService.scala

```diff
@@ -0,0 +1,82 @@
+package com.gu.identity.formstackbatonrequests
+
+import com.typesafe.scalalogging.LazyLogging
+import scalaj.http.Http
+import io.circe.parser.decode
+import com.gu.identity.formstackbatonrequests.aws.SubmissionTableUpdateDate
+import io.circe.Json
+import io.circe.generic.JsonCodec
+
+trait FormstackSar {
+  def accountFormsForGivenPage(page: Int, accountToken: FormstackAccountToken): Either[Throwable, FormstackFormsResponse]
+  def formSubmissionsForGivenPage(page: Int, formId: String, minTime: SubmissionTableUpdateDate, encryptionPassword: String, accountToken: FormstackAccountToken): Either[Throwable, FormstackFormSubmissionsResponse]
+}
+/* Codecs for decoding accountFormsForGivenPage response */
+@JsonCodec case class FormstackForm(id: String)
+@JsonCodec case class FormstackFormsResponse(forms: List[FormstackForm], total: Int)
+
+/* Codecs for decoding formSubmissionsForGivenPage response */
+@JsonCodec case class FormstackResponseValue(value: Json)
+@JsonCodec case class FormstackSubmission(id: String, data: Map[String, FormstackResponseValue])
+@JsonCodec case class FormstackFormSubmissionsResponse(submissions: List[FormstackSubmission], pages: Int)
+
+case class FormstackDecryptionError(message: String) extends Throwable
+
+object FormstackSarService extends FormstackSar with LazyLogging {
+
+  val resultsPerPage = 25
+
+  override def accountFormsForGivenPage(page: Int, accountToken: FormstackAccountToken): Either[Throwable, FormstackFormsResponse] = {
+    val response = Http(s"https://www.formstack.com/api/v2/form.json")
```

Comment:
could we take this URL out into a config? read from parameter store on CF deployment?

Reply:
Definitely can do, there are 2 urls already though and another 2 coming in the next PR so it will be quite a lot of additional config.